### PR TITLE
Learn 23536 homepage functionality

### PR DESCRIPTION
**Pull request by Ashwath-Kaplan**
JIRA : 
https://kna-core.atlassian.net/browse/LEARN-23536
https://kna-core.atlassian.net/browse/LEARN-23537
https://kna-core.atlassian.net/browse/LEARN-23538
https://kna-core.atlassian.net/browse/LEARN-23539

# Summary
- Create toolbar with login and signup buttons.
- Create dialog model for both login and signup.
- Add validators for the input fields in dialog form.
- Add snack bar component.
- Add api services to create new user, login and logout.
- Add signup and login functionality.

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,7 +1,12 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
+import { AppComponent } from './app.component';
 
-const routes: Routes = [];
+const routes: Routes = [
+  { path: '', pathMatch: 'full', component: AppComponent },
+  // { path: 'login', component: LogInComponent },
+  // { path: 'register', component: RegisterComponent }
+];
```
Comment (MuskaaanBansal) [inline src/app/app-routing.module.ts]: remove commented out code from everywhere

File: src/app/components/tool-bar/tool-bar.component.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+.navbar__logo {
+  width: 55px;
+  margin: 5px;
+}
+
+.navbar__links {
+  margin-left: 10px;
+  border: 1px solid black;
+}
```
Comment (MuskaaanBansal) [inline src/app/components/tool-bar/tool-bar.component.scss]:
```suggestion
.navbar {
  background-color: rgb(136,89,83);
  display: flex;
  justify-content: space-between;
  
    &__header{
       color: white;
    }
    
    &__logo {
      width: 55px;
      margin: 5px;
   }
   
   &__links {
    margin-left: 10px;
    border: 1px solid black;
  }

 }
```

**Review by MuskaaanBansal — APPROVED** [review]

File: src/app/components/tool-bar/tool-bar.component.scss (hunk 2 above)
Comment (MuskaaanBansal) [reply src/app/components/tool-bar/tool-bar.component.scss]: css files are not in proper format, please refer to the above changes and update the rest of the files in the same format.

**Comment by MuskaaanBansal** [comment]
Screen Shot 2023-10-03 at 11 58 47 AM

Error messages are not aligned properly

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (Ashwath-Kaplan) [reply src/app/app-routing.module.ts]: removed unnecessary comments. please review it in the new commit.

File: src/app/components/tool-bar/tool-bar.component.scss (hunk 2 above)
Comment (Ashwath-Kaplan) [reply src/app/components/tool-bar/tool-bar.component.scss]: Changed the format of all the scss files to BEM convention.

File: src/app/components/login-dialog/login-dialog.component.scss [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+.login-dialog__fields {
+  display: flex;
+  width: 260px;
+  margin: auto;
+}
+
+// .login-dialog__fields--input {
+//   border: 1px solid black;
+//   border-radius: 4px;
+// }
```
Comment (MuskaaanBansal) [inline src/app/components/login-dialog/login-dialog.component.scss]: remove commented code

File: src/app/components/login-dialog/login-dialog.component.scss [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  font-size: 20px;
+  margin: 20px;
+  cursor: pointer;
+}
+
+.login-dialog__fields {
+  display: flex;
+  width: 260px;
+  margin: auto;
+}
```
Comment (MuskaaanBansal) [inline src/app/components/login-dialog/login-dialog.component.scss]: css file not updated in the correct format

File: src/app/components/login-dialog/login-dialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import { MatDialogRef } from '@angular/material/dialog';
+import { SnackBarService } from '../../services/snack-bar/snack-bar.service';
+
+@Component({
+  selector: 'app-login-dialog',
+  templateUrl: './login-dialog.component.html',
+  styleUrls: ['./login-dialog.component.scss'],
+})
+export class LoginDialogComponent {
+  hide = true;
```
Comment (MuskaaanBansal) [inline src/app/components/login-dialog/login-dialog.component.ts]:
```suggestion
  hidePassword = true;
```

File: src/app/components/tool-bar/tool-bar.component.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  <h1 class="navbar__header">Welcome to Kaplan ReadHub</h1>
+  <div>
+    <a
+      class="navbar__links"
+      mat-button
+      mat-raised-button
+      (click)="openLoginDialog()"
+      >Log in</a
+    >
+    <a
```
Comment (MuskaaanBansal) [inline src/app/components/tool-bar/tool-bar.component.html]:
```suggestion
    <button
```

Comment (MuskaaanBansal) [reply src/app/components/tool-bar/tool-bar.component.html]: make this change every where

File: src/app/components/login-dialog/login-dialog.component.scss [hunk 7]
```diff
@@ -0,0 +1,62 @@
+.login-dialog {
+  width: 300px;
```
Comment (MuskaaanBansal) [inline src/app/components/login-dialog/login-dialog.component.scss]:
```suggestion
  
```

Comment (MuskaaanBansal) [reply src/app/components/login-dialog/login-dialog.component.scss]: add this width while calling dialog box for both sign up and login modals

File: src/app/components/login-dialog/login-dialog.component.scss [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  margin: auto;
+}
+
+// .login-dialog__fields--input {
+//   border: 1px solid black;
+//   border-radius: 4px;
+// }
+
+.login-dialog__fields--error {
+  margin-left: -20px;
```
Comment (MuskaaanBansal) [inline src/app/components/login-dialog/login-dialog.component.scss]:
```suggestion
  ```

Comment (MuskaaanBansal) [reply src/app/components/login-dialog/login-dialog.component.scss]: remove this and use appropriate css class to override the padding

Comment (MuskaaanBansal) [reply src/app/components/login-dialog/login-dialog.component.scss]: make the same change in sign-up as well

File: src/app/components/snack-bar/snack-bar.component.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { Component, Inject } from '@angular/core';
+import { MAT_SNACK_BAR_DATA } from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'app-snack-bar',
+  templateUrl: './snack-bar.component.html',
+  styleUrls: ['./snack-bar.component.scss'],
+})
+export class SnackBarComponent {
+  constructor(@Inject(MAT_SNACK_BAR_DATA) public data: any) {}
```
Comment (MuskaaanBansal) [inline src/app/components/snack-bar/snack-bar.component.ts]:
```suggestion
  constructor(@Inject(MAT_SNACK_BAR_DATA) public snackbar: any) {}
```

File: src/app/components/tool-bar/tool-bar.component.html [hunk 10]
```diff
@@ -0,0 +1,20 @@
+<mat-toolbar class="navbar">
+  <img class="navbar__logo" src="./assets/krh-logo.png" />
+  <h1 class="navbar__header">Welcome to Kaplan ReadHub</h1>
+  <div>
+    <a
```
Comment (MuskaaanBansal) [inline src/app/components/tool-bar/tool-bar.component.html]:
```suggestion
    <button
```

**Pull request merged**